net/sing-box/files/sing-box.init

```diff
@@ -9,13 +9,15 @@ PROG="/usr/bin/sing-box"
 start_service() {
 	config_load "$NAME"
 
-	local enabled user group conffile workdir
+	local enabled user group conffile workdir postup
 	config_get_bool enabled "main" "enabled" "0"
 	[ "$enabled" -eq "1" ] || return 0
 
 	config_get user "main" "user" "root"
 	config_get conffile "main" "conffile"
 	config_get workdir "main" "workdir" "/usr/share/sing-box"
+	config_get postup "main" "postup"
+	[ -x "$postup" ] && ${postup}
```

Comment:
Users can set up additional firewall rules here and still retain them after the upgrade.

Reply:
> > Users can set up additional firewall rules here and still retain them after the upgrade.
> 
> If you want to use TPROXY from an external script, you need to add additional settings in the firewall, because if the firewall reloads, your settings will disappear. The functionality you added is incomplete.
> 
> Refer： https://github.com/openwrt/packages/tree/master/net/miniupnpd/files

The user can also run uci within the script, and it is up to the user to decide exactly what to do.

Reply:
If it is not set, or is not runnable, there will be no additional operation.

Reply:
These commands are supposed to be part of the service, not create a new service to run them.

This was added with reference to wireguard.

Reply:
> > Users can set up additional firewall rules here and still retain them after the upgrade.
> 
> If you want to use TPROXY from an external script, you need to add additional settings in the firewall, because if the firewall reloads, your settings will disappear. The functionality you added is incomplete.
> 
> Refer： https://github.com/openwrt/packages/tree/master/net/miniupnpd/files

I test it, restart firewall only reloads the table inet fw4.
So just use nft add table inet/ip xxx before add rule.